lib/order.rb

```diff
@@ -0,0 +1,38 @@
+
+class Order
+  attr_reader :basket
+
+  def initialize
+    @basket = []
+  end
+
+  def add_dish(dish, price)
+    @basket << { dish => price }
+  end
+
+  def total
+    order_total = 0
+
+    @basket.each do |item|
+      order_total += item.values[0]
+    end
+
+    order_total
+  end
+
+  def show_basket
+    return "Empty basket" if @basket.empty?
+
+    formatted_basket = ""
+
+    @basket.each do |item|
+      name = item.keys[0]
+      price = item[name]
+
+      formatted_basket += "'#{name} $#{format("%.2f", price)}', "
+    end
+
+    "Your order is: #{formatted_basket}total is: $#{format("%.2f", total)}"
```

Comment:
show_basket seems to format *and* show the basket. As I mentioned above these seem to be two different methods. Should they be in the Order class? They seem to be doing the work of a Display class.